src/reader.rs

```diff
@@ -102,19 +100,28 @@ impl Iterator for Reader {
             Endianness::Little => self.reader.read_i32::<LittleEndian>(),
             Endianness::Big => self.reader.read_i32::<BigEndian>()
         } {
-            let mut streamline = Vec::with_capacity(nb_points as usize);
-            let (mut scalars, mut properties) = self.get_sp();
+            let nb_points = nb_points as usize;
+            let mut streamline = Vec::with_capacity(nb_points);
+            let mut scalars = ArraySequence::with_capacity(nb_points * self.nb_scalars);
+            let mut properties = Vec::with_capacity(self.nb_properties);
             match self.endianness {
-                Endianness::Little => self.read_streamline::<LittleEndian>(
-                    &mut streamline, &mut scalars, &mut properties, nb_points as usize),
-                Endianness::Big => self.read_streamline::<BigEndian>(
-                    &mut streamline, &mut scalars, &mut properties, nb_points as usize)
+                Endianness::Little => {
+                    self.read_streamline::<LittleEndian>(
+                        &mut streamline, &mut scalars, nb_points as usize);
+                    for _ in 0..self.nb_properties {
```

Comment:
You have this loop 3 times in the code. I would suggest you put it in a function.

Reply:
One of the uses is for a `ArraySequence` though, but I still created functions for both cases.